### ISSUE TITLE
the problem during training

### ISSUE THREAD

**Issue by xiezhiqing0112** [zju3dv/LoFTR#30]
Thank you for your amazing work. I have some question that the **expec_f_gt** after **spvs_fine() function** are all very large number. And this case cause the **correct_mask** in computing fine loss is zero.

**Comment by angshine** [comment]
Thank you for your interest in our work. I can't position the problem without further information. On which dataset are you training? Which matcher are you using? When did you meet this problem during the entire training phase? Did it happen from the very start? A training sample with an `expec_f_gt` value bigger than 1.0 means an invalid sample for the fine-level window used, which would be filtered during the loss calculation. However, we always take parts of the fine-level training samples from the gt coarse correspondences, aiming at alleviating this problem.

**Comment by xiezhiqing0112** [comment]
I am training in Megadepth dataset. the matcher is dual softmax. this problem happen from the start. I don't sure it can be improved after several epochs. and the coarse loss weight and fine loss weight are all **1**.  Does this case make training unstable?

**Comment by angshine** [comment]
Does the training converge with an increasing pose AUC or the training never lead to the pose AUC boosting? At which image resolution are you training? What pose AUC could you get?

**Comment by xiezhiqing0112** [comment]
I don't get the AUC result now. but during training, the logging always appear **assign a false supervision to avoid ddp deadlock.** the image resolutiong is 640

**Comment by angshine** [comment]
What `batch_size` are you using? I could try to reproduce your problem and give you more feedback then.

**Comment by xiezhiqing0112** [comment]
8.

**Comment by xiezhiqing0112** [comment]
hello. Maybe **warp_kpts function** has some problem.I vis the keypoint0 and w_keypoint0.,but the match is wrong. But I don't sure. whether you are in the same case.

**Comment by angshine** [comment]
Hi,  I tried to train LoFTR in your set-up with the released code, and I could not reproduce your problem. Note that I update the set-up of the MegaDepth depth map [here](https://github.com/zju3dv/LoFTR/blob/master/docs/TRAINING.md), please make sure you have set up the dataset correctly.

**Comment by xiezhiqing0112** [comment]
I use **d2-net** to preprocess the **medadepth** dataset. Maybe this make camera poses different. Could you share the data preprocess code for me?

**Comment by angshine** [comment]
Hi, I forgot to mention that we use the [d2-net preprocessed images and camera poses](https://drive.google.com/drive/folders/1hxpOsqOZefdrba_BqnW490XpNX_LgXPB) as well during training (I will update the doc asap). In addition, the original depth maps from [MegaDepth](https://www.cs.cornell.edu/projects/megadepth/dataset/Megadepth_v1/README.md) are used.

**Comment by xiezhiqing0112** [comment]
thank you very much!!
